crates/miden-testing/src/executor.rs

```diff
@@ -37,14 +37,25 @@ impl<H: Host> CodeExecutor<H> {
         self
     }
 
-    /// Compiles and runs the desired code in the host and returns the [Process] state
+    /// Compiles and runs the desired code in the host and returns the [`Process`] state.
+    ///
+    /// To improve the error message quality, convert the returned [`ExecutionError`] into a
+    /// [`Report`](miden_objects::assembly::diagnostics::Report).
     pub fn run(self, code: &str) -> Result<Process, ExecutionError> {
-        let assembler = TransactionKernel::testing_assembler();
+        let assembler = TransactionKernel::testing_assembler().with_debug_mode(true);
```

Comment:
Question: should `TransactionKernel::testing_assembler()` already return the assembler in debug mode?

Reply:
I think so, yes, but on the other hand these `with_debug_mode` calls will all go away when https://github.com/0xMiden/miden-vm/issues/1821 is implemented. I'll add it to `testing_assembler` and leave the existing overrides in place, because it's probably not worth changing it now.